.eslintrc.json

```diff
@@ -1,14 +1,21 @@
 {
-  "extends": ["next/core-web-vitals", "eslint:recommended", "plugin:@typescript-eslint/recommended", "prettier"],
+  "extends": [
+    "next/core-web-vitals",
+    "eslint:recommended",
+    "plugin:@typescript-eslint/recommended",
+    "prettier"
+  ],
   "parser": "@typescript-eslint/parser",
   "parserOptions": {
     "ecmaVersion": "latest",
     "sourceType": "module",
     "project": ["./tsconfig.json"]
   },
   "plugins": ["@typescript-eslint", "prettier", "eslint-plugin-tsdoc"],
+  "ignorePatterns": ["generated"],
   "rules": {
-    "@typescript-eslint/no-unsafe-argument": "error",
+    // TODO discuss removing this rule
```

Comment:
I deleted this rule because it was causing major type issues with my newer auth logic. It turns out typechecks actually pass with actual compiling, it's just the typescript language server that has issues with my code, so I will undo the removal and make a complaint to vscode instead.